lib/src/search_page.dart

```diff
@@ -0,0 +1,141 @@
+import 'package:flutter/material.dart';
+
+import '../consts.dart';
+import 'countries.dart';
+
+class SearchPage extends StatefulWidget {
+  const SearchPage({super.key});
+
+  @override
+  State<SearchPage> createState() => _SearchPageState();
+}
+
+class _SearchPageState extends State<SearchPage> {
+  List<Map<String, String>> allCountries = Countries.allCountries;
+  List<Map<String, String>> filteredCountries = [];
+  void initState() {
+    super.initState();
+    filteredCountries = allCountries; // Initialize with all countries
+  }
+
+  void updateFilteredCountries(List<Map<String, String>> countries) {
+    setState(() {
+      filteredCountries = countries;
+    });
+  }
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text("Select a Country"),
+      ),
+      body: Column(
+        children: [
+          CountrySearchBar(
+            allCountries: allCountries,
+            onSearch: updateFilteredCountries,
+          ),
+          Expanded(
+            child: ListView.builder(
+              itemCount: filteredCountries.length,
+              itemBuilder: (context, index) {
+                final country = filteredCountries[index];
+                return GestureDetector(
+                  onTap: () {
+                    // Pass the selected country back to the previous page
+                    Navigator.pop(context, country);
+                  },
+                  child: Container(
+                    padding: EdgeInsets.symmetric(vertical: 10, horizontal: 16),
+                    child: Row(
```

Comment:
You could give horizontal padding to the row as they feel they are a bit to the extreme.

Reply:
20 would be fine?

Reply:
Anything would be fine, just it should look good